### ISSUE TITLE
SmartCity-Accessibility Server Side Unit Tests #844 failed

### ISSUE THREAD

**Issue by JuanJenkinsBot** [TechnionYP5777/SmartCity-Accessibility#425]
Build 'SmartCity-Accessibility Server Side Unit Tests' is failing!

Last 50 lines of build :
```
[...truncated 169.53 KB...]
java.lang.NullPointerException
	at smartcity.accessibility.database.UserProfileManagerTest.setUpMock(UserProfileManagerTest.java:133)
	at smartcity.accessibility.database.UserProfileManagerTest.setUp(UserProfileManagerTest.java:46)

testMostHelpful(smartcity.accessibility.database.UserProfileManagerTest)  Time elapsed: 0.025 sec  <<< ERROR!
java.lang.NullPointerException
	at smartcity.accessibility.database.UserProfileManagerTest.setUpMock(UserProfileManagerTest.java:133)
	at smartcity.accessibility.database.UserProfileManagerTest.setUp(UserProfileManagerTest.java:46)

testUpdate(smartcity.accessibility.database.UserProfileManagerTest)  Time elapsed: 0.025 sec  <<< ERROR!
java.lang.NullPointerException
	at smartcity.accessibility.database.UserProfileManagerTest.setUpMock(UserProfileManagerTest.java:133)
	at smartcity.accessibility.database.UserProfileManagerTest.setUp(UserProfileManagerTest.java:46)

Running smartcity.accessibility.search.SearchTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 1.094 sec - in smartcity.accessibility.search.SearchTest

Results :

Tests in error: 
  UserProfileManagerTest.setUp:46->setUpMock:133 » NullPointer
  UserProfileManagerTest.setUp:46->setUpMock:133 » NullPointer
  UserProfileManagerTest.setUp:46->setUpMock:133 » NullPointer
  UserProfileManagerTest.setUp:46->setUpMock:133 » NullPointer
  UserProfileManagerTest.setUp:46->setUpMock:133 » NullPointer
  UserProfileManagerTest.setUp:46->setUpMock:133 » NullPointer
  UserProfileManagerTest.setUp:46->setUpMock:133 » NullPointer
  UserInformationServiceTest.searchSuccess:45 » NestedServlet Request processing...

Tests run: 77, Failures: 0, Errors: 8, Skipped: 2

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:25 min
[INFO] Finished at: 2017-06-08T15:54:57+00:00
[INFO] Final Memory: 34M/183M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project SmartCity-Accessibility: There are test failures.
[ERROR] 
[ERROR] Please refer to /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Build step 'Invoke top-level Maven targets' marked build as failure
```

Changes since last successful build:
- [yaelAmitay] 579018450455d1e7c1107fe0b6dc18c412ebb8e0 - add image to parse #424

[View full output](http://kapalex.ddns.net:8080/job/SmartCity-Accessibility%20Server%20Side%20Unit%20Tests/844/)

**Comment by JuanJenkinsBot** [comment]
Build 'SmartCity-Accessibility Server Side Unit Tests' is failing!

Last 50 lines of build :
```
[...truncated 168.25 KB...]
    {"numOfReviews" = 5, "rating" = 25, "username" = "alexaxa"}
);
-> at smartcity.accessibility.database.UserProfileManagerTest.testPut(UserProfileManagerTest.java:80)
Actual invocation has different arguments:
database.put(
    "UserProfile",
    {"numOfReviews" = 5, "rating" = 25, "profileImg" = null, "username" = "alexaxa"}
);
-> at smartcity.accessibility.database.UserProfileManager.lambda$put$1(UserProfileManager.java:129)

  UserProfileManagerTest.testUpdate:87 
Argument(s) are different! Wanted:
database.update(
    "UserProfile",
    "MY_ID",
    {"numOfReviews" = 5, "rating" = 25, "username" = "alexaxa"}
);
-> at smartcity.accessibility.database.UserProfileManagerTest.testUpdate(UserProfileManagerTest.java:87)
Actual invocation has different arguments:
database.update(
    "UserProfile",
    "MY_ID",
    {"numOfReviews" = 5, "rating" = 25, "profileImg" = null, "username" = "alexaxa"}
);
-> at smartcity.accessibility.database.UserProfileManager.lambda$update$2(UserProfileManager.java:149)

Tests in error: 
  UserInformationServiceTest.searchSuccess:45 » NestedServlet Request processing...

Tests run: 77, Failures: 2, Errors: 1, Skipped: 2

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:24 min
[INFO] Finished at: 2017-06-08T16:03:04+00:00
[INFO] Final Memory: 34M/214M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project SmartCity-Accessibility: There are test failures.
[ERROR] 
[ERROR] Please refer to /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Build step 'Invoke top-level Maven targets' marked build as failure
```

Changes since last successful build:
- [yaelAmitay] 579018450455d1e7c1107fe0b6dc18c412ebb8e0 - add image to parse #424

- [yaelAmitay] 0a440d3e41b809afaf6b5b00f08eb13bfeb06f4a - fix null get image bug #424



[View full output](http://kapalex.ddns.net:8080/job/SmartCity-Accessibility%20Server%20Side%20Unit%20Tests/845/)

**Comment by JuanJenkinsBot** [comment]
Build 'SmartCity-Accessibility Server Side Unit Tests' is failing!

Last 50 lines of build :
```
[...truncated 164.32 KB...]
Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.09 sec - in smartcity.accessibility.socialnetwork.ReviewCommentTest
Running smartcity.accessibility.socialnetwork.HelpfulnessTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec - in smartcity.accessibility.socialnetwork.HelpfulnessTest
Running smartcity.accessibility.socialnetwork.UserFunctionalityTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.468 sec - in smartcity.accessibility.socialnetwork.UserFunctionalityTest
Running smartcity.accessibility.socialnetwork.BestReviewsTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 sec - in smartcity.accessibility.socialnetwork.BestReviewsTest
Running smartcity.accessibility.socialnetwork.UserPinTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.201 sec - in smartcity.accessibility.socialnetwork.UserPinTest
Running smartcity.accessibility.socialnetwork.DeleteReviewsTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.298 sec - in smartcity.accessibility.socialnetwork.DeleteReviewsTest
Running smartcity.accessibility.socialnetwork.ScoreTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.78 sec - in smartcity.accessibility.socialnetwork.ScoreTest
Running smartcity.accessibility.database.UserManagerTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.762 sec - in smartcity.accessibility.database.UserManagerTest
Running smartcity.accessibility.database.LocationManagerTest
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.351 sec - in smartcity.accessibility.database.LocationManagerTest
Running smartcity.accessibility.database.ReviewManagerTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.933 sec - in smartcity.accessibility.database.ReviewManagerTest
Running smartcity.accessibility.database.UserProfileManagerTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.821 sec - in smartcity.accessibility.database.UserProfileManagerTest
Running smartcity.accessibility.search.SearchTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 1.007 sec - in smartcity.accessibility.search.SearchTest

Results :

Tests in error: 
  UserInformationServiceTest.searchSuccess:45 » NestedServlet Request processing...

Tests run: 77, Failures: 0, Errors: 1, Skipped: 2

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:19 min
[INFO] Finished at: 2017-06-08T16:16:45+00:00
[INFO] Final Memory: 34M/215M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project SmartCity-Accessibility: There are test failures.
[ERROR] 
[ERROR] Please refer to /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Build step 'Invoke top-level Maven targets' marked build as failure
```

Changes since last successful build:
- [yaelAmitay] 579018450455d1e7c1107fe0b6dc18c412ebb8e0 - add image to parse #424

- [yaelAmitay] 0a440d3e41b809afaf6b5b00f08eb13bfeb06f4a - fix null get image bug #424

- [yaelAmitay] 2913a9629d5854377894eded30438fc0728c1233 - try fixing the bug #424



[View full output](http://kapalex.ddns.net:8080/job/SmartCity-Accessibility%20Server%20Side%20Unit%20Tests/846/)

**Comment by JuanJenkinsBot** [comment]
Build was fixed!
